### DISCUSSION TITLE
Widevine Support Considerations

### DISCUSSION THREAD

**Discussion by burnout426** [Ld-Hagen/fix-opera-linux-ffmpeg-widevine#4] [General]
In fix-opera.sh, I see you already put the WidevineCDM folder for Opera in the lib_extra folder so that Opera has its own copy somewhere and modify widevine_config.json to search in that WidevineCDM folder. Just note though that it used to be the case that widevine_config.json would be overwritten on update of Opera and you'd have to re-do changes to the file after each Opera update. So, I would double-check that the file survives an update.

Note though that since Opera 76.0.3981.0, Opera doesn't come with a widevine_config.json file by default because it's not needed. Instead, Widevine is automatically fetched and the WidevineCDM folder is put in the Opera profile folder that's in /home/username/.config. You can check and update Widevine by going to the URL opera://components in Opera. To refetch Widevine, you can delete that WidevineCDM folder (while Opera is closed) and go back to opera://components and click "update".

With that said, it might not be necessary to even worry about Widevine. However, if you still want to anyway, it might be better if the script checked for Widevine in the Opera users's profile folder and only put a new one there if there isn't one (as the user, not sudo). Something to think about. You might also want to check that Opera still uses widevine_config.json if you put it there.

**Comment by Ld-Hagen** [comment]
> Just note though that it used to be the case that widevine_config.json would be overwritten on update of Opera and you'd have to re-do changes to the file after each Opera update. So, I would double-check that the file survives an update.

The script is made to be ran after each opera update. Manually or using auto-start hook for apt.


> Note though that since Opera 76.0.3981.0, Opera doesn't come with a widevine_config.json file by default because it's not needed. Instead, Widevine is automatically fetched and the WidevineCDM folder is put in the Opera profile folder that's in /home/username/.config.

It didn't work for me for some reason (at least on Opera 78 builds). While placing it to Opera's lib_extra dir works fine. But probably I'll check it again.
And also Widevine installation is optional an may be easily disabled by setting  **FIX_WIDEVINE=false**

**Comment by Ld-Hagen** [comment]
Well, I've just tested it again. For some reason Widevine from opera profile didn't work for me. Deleting it and then refetching helped. But I don't find it a user-friendly way, especially if there more than one user on PC.

For those who doesn't want to use this Widevine fix I've added instruction to disable it.

**Comment by F4VSJ** [comment]
this solution worked for me
debian 11 and Opera Version:82.0.4227.23
Thanks
